src/test/scala/org/zalando/zhewbacca/IAMClientSpec.scala

```diff
@@ -101,7 +101,7 @@ class IAMClientSpec extends Specification {
     }
 
     "should not return a Token Info when remote server operates slow" in {
-      val app: Application = fakeApp(delay = 15.minutes)
+      val app: Application = fakeApp(delay = 5.seconds)
```

Comment:
The tests behave different now. With fakeApp(delay = 15.minutes) the tests really wait for 15 minutes to complete.

Reply:
Hm, strange. It can be that circuit breaker does not apply. I'll have a look.

Reply:
But the test should fail then, right? The test finishes successfully but afterwards the ```Thread.sleep``` in 

```scala
def fakeApp(delay: Duration = 0.second, response: Result = Results.Ok): Application = {
    val routes: PartialFunction[(String, String), Handler] = {
      case ("GET", "/tokeninfo") => Action {
        Thread.sleep(delay.toMillis)
        response
      }
    }
```

seems to block the execution of the remaining tests. This behaviour seems to be different compared to the 2.5.x version.

Reply:
The intention of having such code was to reproduce slow operation and library reaction on it. 

Probably now it blocks test executor's thread, so I need to find another way to achieve the same behaviour. For me it looks a little bit wired because I assume when we start a new app it has to have own thread pool to serve request, so test thread shouldn't be blocked.

Reply:
Yep, I also was surprised...